src/board_controller/muse/muse2.cpp

```diff
@@ -0,0 +1,399 @@
+#include "muse2.h"
+#include "muse_constants.h"
+
+#include <iostream>
+#include <stdexcept>
+
+using json = nlohmann::json;
+
+Muse2::Muse2 (struct BrainFlowInputParams params) : Board ((int)BoardIds::MUSE_2_BOARD, params)
+{
+    is_streaming = false;
+    keep_alive = false;
+    initialized = false;
+}
+
+Muse2::~Muse2 ()
+{
+    skip_logs = true;
+    release_session ();
+}
+
+int Muse2::prepare_session ()
+{
+    safe_logger (spdlog::level::trace, "prepare session");
+    if (initialized)
+    {
+        safe_logger (spdlog::level::info, "Session is already prepared");
+        return (int)BrainFlowExitCodes::STATUS_OK;
+    }
+
+    if (bt.open_bt_adapter () != (int)BluetoothExitCodes::OK)
+    {
+        // Error codes could be processed better.
+        // The current error code selections in this sourcefile could be considered stubs if
+        // insufficient.
+        return (int)BrainFlowExitCodes::UNABLE_TO_OPEN_PORT_ERROR;
+    }
+
+    // find and connect to device
+
+    if (params.mac_address.empty ())
+    {
+        std::cerr << "No mac address provided, scanning for a nearby muse ..." << std::endl;
```

Comment:
use logger instead std::cerr

Reply:
i'm just trying to get it up and running now, so focusing my time on what makes it run

Reply:
this seems really low priority to me, given i don't have too much time to add here.  it seems what is needed here is adding more muse bluetooth support, and resolving the bglib/bluetooth.cpp difference.